### ISSUE TITLE
Retract Notifications if No Longer Able to Participate

### ISSUE THREAD

**Issue by youralien** [NUDelta/ce-platform#46]
Kapil: 

> Critical incident! I’m currently commuting up on lakeshore drive and got a Notification to participate in half half travel. But my location updated before I could participate, so when I go into the app all I see is a blank screen

**Comment by kapil1garg** [comment]
One fix to try is as follows:
- For every function that resets availability or assignments in the `onLocationUpdate` function, first check to see if the new affordances are the same as before. One way to check this is to see if users can still match the needs given their new location and affordances. If so, then we shouldn't pull their availability since it's still valid
- Any places where we try to use location data as a decision variable (ex. in timeouts when starting the coordinator, we look to see if new location is <= 10.0 meters away), check if the affordances match instead since CE is really only thinking about affordances and not actual location. This should allow for cases where a user might be walking through a park to be notified anytime they are in the park since their affordances still match the park (whereas now, they wouldn't be able to participate unless they were in the park AND relatively stationary due to the 10m threshold in timeouts).

See below for some of the sketches we made regarding this issue.
[Whiteboard.pdf](https://github.com/NUDelta/ce-platform/files/2237273/Whiteboard.pdf)

@youralien will work on generating simulations of potential scenarios that users may go though (ex. walking to a place, stopping and getting matched with need, walking again; walking through a large place like a park where affordance is always valid) so we can test the changes against that.

**Comment by youralien** [comment]
TL;DR:  Tried to change the interface model so that notifications for OCEs would be cleared or retracted.  However, the package call for `clearAllNotifications` does nothing.  Attempted changes [at this commit](https://github.com/NUDelta/ce-platform/commit/76547095f65c30c0323fdbbc600cb5fca6bcb55c) in case anyone wants to pick up where I left off.

Other possible change to the interface model is to send another notification, apologizing that they missed the opportunity.  Right now, the home screen has an apology, but no one is seeing that.

**Comment by youralien** [comment]
I am going to keep this open.  However, #78 worked on parts of this issue. Namely,

* User Interface Fix: Sends users a notification if they missed a chance to participate
* System Fix: User will get notified for new experiences, if they missed a change to participate. They won't be limited by the last time they got a notification.
